### PR TITLE
fix: merge frame url and frame url fragment for getting full url

### PR DESCRIPTION
**Pull request by bmlyr**
This Pr : 

- Retrieve full url by adding url fragment to url when trying to call url function from a frame.

### CODE REVIEW

**Comment by bmlyr** [comment]
@googlebot I signed it!

**Comment by bryanmlyr** [comment]
@googlebot I fixed it.

**Comment by bmlyr** [comment]
@googlebot I fixed it.

**Comment by mathiasbynens** [comment]
Thanks for the patch. Two requests:

- In the commit message, please explain what problem this is fixing.
- Please add a test

**Comment by bmlyr** [comment]
> Thanks for the patch. Two requests:
> 
> * In the commit message, please explain what problem this is fixing.
> * Please add a test

Done 😄

File: test/assets/frames/one-frame-url-fragment.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<iframe src='./frame.html&test=fragment'></iframe>
```
Comment (mathiasbynens) [inline test/assets/frames/one-frame-url-fragment.html]:
```suggestion
<iframe src='./frame.html&test=fragment'></iframe>

```

**Review by mathiasbynens — APPROVED** [review]

**Comment by mathiasbynens** [comment]
@mjzffr FYI, the CI has been unhappy lately, with Firefox launch failing. Does this ring any bells?

**Comment by mathiasbynens** [comment]
Ah, this is still #5673. Will post there

File: test/assets/frames/one-frame-url-fragment.html (hunk 1 above)
Comment (campersau) [inline test/assets/frames/one-frame-url-fragment.html:1]: Shouldn't this be `./frame.html?test=fragment` or `./frame.html#fragment` instead?
According to https://github.com/puppeteer/puppeteer/blob/main/new-docs/puppeteer.protocol.page.frame.urlfragment.md the fragment is `#fragment` and `?param=value` is often referred to as query.

Comment (mathiasbynens) [reply test/assets/frames/one-frame-url-fragment.html:1]: @campersau That's a good point. Happy to accept a PR that changes these names!

**Pull request merged**